timemachine/fe/single_topology.py

```diff
@@ -1158,81 +906,105 @@ def assert_torsions_defined_over_non_linear_angles(system: GuestSystem | HostGue
                 )
 
 
-def assert_chiral_consistency(src_chiral_idxs: NDArray, dst_chiral_idxs: NDArray):
-    """
-    Assert that the chiral volumes are not inverting in src and dst.
-    """
-    src_chiral_restr_idx_set = ChiralRestrIdxSet(src_chiral_idxs)
-    dst_chiral_restr_idx_set = ChiralRestrIdxSet(dst_chiral_idxs)
-
-    # ensure that we don't have any chiral inversions between src and dst end states
-    assert len(src_chiral_restr_idx_set.allowed_set.intersection(dst_chiral_restr_idx_set.disallowed_set)) == 0
-    assert len(dst_chiral_restr_idx_set.allowed_set.intersection(src_chiral_restr_idx_set.disallowed_set)) == 0
-
+def build_combined_atom_map_mixin(
+    num_host_atoms: int, mol_a: Chem.Mol, mol_b: Chem.Mol, core: NDArray[np.int32]
+) -> AtomMapMixin:
+    new_core_ = []
+    for idx in range(num_host_atoms):
+        new_core_.append((idx, idx))
 
-@dataclass
-class AlignedPotential:
-    idxs: NDArray[np.int32]
-    src_params: NDArray[np.float64]
-    dst_params: NDArray[np.float64]
-    mins: NDArray[np.float64]
-    maxes: NDArray[np.float64]
+    for i, j in core:
+        new_core_.append((i + num_host_atoms, j + num_host_atoms))
 
-    def interpolate(self, lamb):
-        raise NotImplementedError()
+    new_core = np.array(new_core_, dtype=np.int32)
 
+    num_atoms_a = num_host_atoms + mol_a.GetNumAtoms()
+    num_atoms_b = num_host_atoms + mol_b.GetNumAtoms()
 
-class AlignedBond(AlignedPotential):
-    def interpolate(self, lamb):
-        k_min = 0.1
-        params = batch_interpolate_harmonic_bond_params(
-            self.src_params, self.dst_params, lamb, k_min, self.mins, self.maxes
-        )
-        params = jnp.array(params).T
+    return AtomMapMixin(num_atoms_a, num_atoms_b, new_core)
 
-        return HarmonicBond(self.idxs).bind(params)
 
+def build_end_state_reference_all_pairs(
```

Comment:
Might add a short docstring? (What does "reference" refer to here?)

Reply:
Renamed `build_end_state_reference_all_pairs` -> `build_end_state_nonbonded_all_pairs` and `build_end_state_reference_ixn_group` -> `build_end_state_nonbonded_ixn_group`. Will add a docstring.

My intention is for this to ultimately go into the builders class with the next refactor.